cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -671,6 +678,22 @@ private void emitProvisioningTimeMetric(ProgramRunId programRunId, ProfileId pro
                                                     provisioningTime);
   }
 
+  /**
+   * Emit the pipeline run time metric. The tags are constructed with the program run id and program run status.
+   */
+  private void emitRunTimeMetric(ProgramRunId programRunId, ProgramRunStatus programRunStatus,
+                                 long runTime) {
+    Map<String, String> tags = ImmutableMap.<String, String>builder()
+      .put(Constants.Metrics.Tag.NAMESPACE, programRunId.getNamespace())
+      .put(Constants.Metrics.Tag.APP, programRunId.getApplication())
```

Comment:
Only having app in the context meaning we can't distinguish different program run time in the metrics? I think the program should be in the context hierarchy

Reply:
fixed.